pkg/api/http.go

```diff
@@ -810,3 +812,19 @@ func ReadBodyToBuffer(resp *http.Response) (*bytes.Buffer, error) {
 
 	return buffer, nil
 }
+
+// ParseTraceLookupRequest takes an http.Request and decodes the body to create a tempopb.TraceLookupRequest
+func ParseTraceLookupRequest(r *http.Request) (*tempopb.TraceLookupRequest, error) {
+	body, err := io.ReadAll(r.Body)
```

Comment:
`io.ReadAll` is memory inefficient. if we can get a body size from the http request we should use [ReadAllWithEstimate](https://github.com/grafana/tempo/blob/97a244012fb84feaf2b75eb9e9b2811ddf57292b/pkg/io/read.go#L10).

An even better solution would be to unmarshal directly from the io.Reader (r.Body) if possible.